cineast-core/src/main/java/org/vitrivr/cineast/core/features/OCRSearch.java

```diff
@@ -1,24 +1,524 @@
 package org.vitrivr.cineast.core.features;
 
+import boofcv.io.image.ConvertBufferedImage;
+import boofcv.struct.image.GrayU8;
+import georegression.struct.shapes.Quadrilateral_F64;
+import org.apache.commons.text.similarity.JaroWinklerSimilarity;
+import org.opencv.core.CvType;
+import org.opencv.core.Mat;
+import org.opencv.core.Point;
+import org.vitrivr.cineast.core.data.Pair;
+import org.vitrivr.cineast.core.data.entities.SimpleFulltextFeatureDescriptor;
+import org.vitrivr.cineast.core.data.segments.SegmentContainer;
 import org.vitrivr.cineast.core.features.abstracts.AbstractTextRetriever;
+import org.vitrivr.cineast.core.util.*;
+
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.util.*;
+import java.util.List;
 
 /**
- *  OCR is handled by adding fuzziness / levenshtein-distance support to the query if there are no quotes present (as quotes indicate precision)
- *  This makes sense here since we expect small errors from OCR sources
+ * OCR is handled by adding fuzziness / levenshtein-distance support to the query if there are no quotes present (as quotes indicate precision) This makes sense here since we expect small errors from OCR sources
  */
 public class OCRSearch extends AbstractTextRetriever {
 
   public static final String OCR_TABLE_NAME = "features_ocr";
 
   /**
-   * Default constructor for {@link OCRSearch}.
+   * Configurations
+   * rate:
+   * Rate refers to the rate at which detections are done. E.g.: Rate = 3 --> Detections are done at every third frame
+   * Increase rate for better inference times
+   *
+   * threshold_CIoU:
+   * CIoU stands for Combined Intersection over Union
+   * When the CIoU is below 0.4, they are regarded as separate streams
+   * Should not be changed
+   *
+   * threshold_postproc: This is the threshold for the postprocessing stream association step
+   * Strongly urge not to change
+   *
+   * tracker_type: Refers to the tracker which is used
+   * threshold_stream_length: Refers to the amount of consecutive frames a text should minimally appear in
+   * If a text appears in less consecutive frames than the threshold, the text is discarded
    */
+  private static final int rate = 3;
+  private static final int batchSize=16;
```

Comment:
Notable change since the last review: I enabled batch processing for the text detector. It is set to 16 for now, but can easily be changed by altering this parameter.